pandas/core/arrays/categorical.py

```diff
@@ -2680,15 +2680,9 @@ def _replace(self, *, to_replace, value, inplace: bool = False) -> Self | None:
         NDArrayBacked.__init__(cat, new_codes, new_dtype)
 
         if new_dtype != orig_dtype:
-            warnings.warn(
-                # GH#55147
+            raise TypeError(
```

Comment:
is this method called anywhere?  i think it can just be removed

Reply:
there is special-casing in internals.blocks that calls this method.  that special-casing needs to be removed